rfc3164/rfc3164.go

```diff
@@ -169,16 +171,21 @@ func (p *Parser) parseTimestamp() (time.Time, error) {
     }
 
     sub = p.buff[p.cursor : tsFmtLen+p.cursor]
-    ts, err = time.ParseInLocation(tsFmt, string(sub), time.Local)
-    if err == nil {
-      /* Set Year on the Timestamp before converting to UTC so that time zone and
-         DST settings (which are year dependent) can be properly assessed in the
-         conversion. */
-      p.fixTimestampIfNeeded(&ts)
-
-      ts = ts.UTC()
-      found = true
-      break
+
+    // Only test the date pattern if the next character is a space (or the end of the message)
+    // This prevents us picking up the first 4 numbers of a hostname as a year
+    if p.cursor+tsFmtLen == p.l || p.buff[p.cursor + tsFmtLen] == ' ' {
```

Comment:
Devil's advocate: what if the host name is entirely numeric?

There's a slightly better chance of this occurring since syslog defaults to the shortname rather than the FQDN.  So a hostname of `1234.prod-web.mydomain.com` will show up as `1234`.  Is it maybe worth checking that the year value is reasonable?  I don't think anyone will have logs with the year 1234 AD.